### DISCUSSION TITLE
Default Page for Pagination with an empty collection

### DISCUSSION THREAD

**Discussion by cmcknight** [11ty/eleventy#2078] [Q&A]
Hi all,

When using pagination, is there some method for generating a default page if a collection is empty?

**Comment by pdehaan** [comment]
See https://github.com/11ty/eleventy/issues/1185 for some suggestions.

**Reply by cmcknight** [reply]
Those didn't really do the trick for me. The template is skipped because the collection is empty. However, I did note that Zach said something about making it a feature request.

**Reply by pdehaan** [reply]
Might be able to fake it w/ two templates (one that paginates+writes files if the data source has items, and another that writes a file only if the data source is empty).

Global data file:
```js
// /_data/cats.js
module.exports = [
  // {name: "Garfield"}
];
```

Nunjucks template which loops over previous global data file (but could be a collection too probably):
```njk
---
# /cats.pagination.njk
pagination:
  data: cats
  size: 1
  alias: cat
permalink: "/cats/{{ cat.name | slug }}/"
---

<h1>{{ cat.name }}</h1>
```

Now, this won't render anything if the `cats` array/collection is empty (which ours is).

But if we create a new template:

```njk
---js
{
  "//": "/cats.pagination.empty.njk",
  title: "No cats found!",
  eleventyComputed: {
    permalink(data) {
      return (data.cats.length === 0) ? "/cats/" : false;
    }
  }
}
---

<h1>{{ title }}</h1>
```

This template uses JavaScript frontmatter to conditionally write the template only if the `cats` array/collection is zero items. Otherwise the `permalink: false` will prevent the file from being written.

If you don't like the JavaScript front matter approach, you could also use a template data file like "cats.pagination.empty.11tydata.js" which exports the `eleventyComputed` object.

**Reply by pdehaan** [reply]
Which I guess is what Zach was saying in https://github.com/11ty/eleventy/issues/1185#issuecomment-629690050 re: creating a separate notes-empty.liquid template.

Although I had problems w/ using something like `permalink: "false"` since the YAML+Nunjucks frontmatter was converting that to a string which was writing "_site/false" which is why I used `eleventyComputed` and a more JavaScript approach.

**Reply by cmcknight** [reply]
My workaround for now, was to drop in an empty 'product' file with a sentinel value indicating there are no products. It works, it feels gross, but it works. I added to #731 a proposed enhancement:

```
pagination:
    data: collection.myCoolStuff
    size: 25
    alias: stuff
    renderEmptyCollections: true | false
```

where true forces the template to be rendered and false is the default behavior (not rendering empty collections).

The developer can add custom text using a conditional to test the length of the collection, i.e.,

```
{% if stuff | length < 1 %}
    <h3>Ain't nuttin' here bruddah!</h3>
{% else %}
  rest of body
{% endif %}
```

I'm still looking for where this would need to happen in the Eleventy source code during my copious spare time although it looks it would be some place in Template.js (maybe)...
